### PR TITLE
24 cpv3 test cases absorption curve

### LINKED ISSUE #24: CPv3 - Test Cases - Absorption Curve
**Description**
---
In Carbon Protocol v3 - implement units tests to coverage multi-scenarios of setting and updated the Carbon Absorption Curve.

The function to tests from [IAbsorber](https://github.com/carbonable-labs/carbon-protocol-v3/blob/main/src/components/absorber/interface.cairo) :
- [ ] set_project_value
- [ ]  set_absorptions

Verify the result is expected for :
- [ ] get_absorptions
- [ ] get_current_absorption
- [ ] get_project_value

**Additional context**
---
- You can find inspiration of tests in [Carbon Protocol v2](https://github.com/Carbonable/carbon-protocol/blob/main/src/tests/test_project.cairo)

- You can find on Cairo OpenZepplin, the kind of testing coverage we want for their ERC-1155, but we want to implement tests with Cairo Foundry,  
[you can find it her](https://github.com/OpenZeppelin/cairo-contracts/blob/v0.10.0/src/tests/token/test_erc1155.cairo)

### PR DESCRIPTION
**Pull request by julienbrs**
closes #24

### CODE REVIEW

**Comment by julienbrs** [comment]
Due to the refactoring made recently, acceptance criteria may have changed.

## Functions tested:
- set_project_carbon
- set_absorptions
- get_current_absorption
### Other minor functions used (getters):
- get_project_carbon
- get_absorptions
- get_times

### Additional changed:
- tests of `test_project.cairo` have been updated and

**Comment by julienbrs** [comment]
Due to the refactoring made recently, acceptance criteria may have changed.

## Functions tested
in `test_carbon_handler.cairo`
- set_project_carbon
- set_absorptions
- get_current_absorption
### Other minor functions used (getters):
- get_project_carbon
- get_absorptions
- get_times

### Additional changed:
- tests of `test_project.cairo` have been updated
- `Absorber_ton_equivalent` has been removed from Storage, and `get_ton_equivalent` now using `CREDIT_CARBON_TON` to convert.
**I feel like  `CREDIT_CARBON_TON` and  `CC_DECIMALS` are redundant; am I wrong?**

File: tests/test_carbon_handler.cairo [hunk 1]
```diff
@@ -109,36 +111,33 @@ fn test_set_project_value() {
...
-    assert(fetched_value == 0, 'default project_value is not 0');
+    // [Assert] default project_carbon is 0
+    let fetched_value = project.get_project_carbon();
+    assert(fetched_value == 0, 'default project_carbon is not 0');
 }
 
 #[test]
-fn test_set_project_value_twice() {
-    // [Setup]
+fn test_set_project_carbon_twice() {
```
Comment (cloudvenger) [inline tests/test_carbon_handler.cairo:123]: maybe change the name to be more behavior oriented by `test_set__and_update_project_carbon`

**Review by cloudvenger — APPROVED** [review]
Really good PR ! 
Just a comment for naming , but can still be merged

**Pull request merged**